resolver/graph.go

```diff
@@ -370,29 +368,51 @@ func validateMessageRuleGraph(graph *MessageRuleDependencyGraph) error {
 	return nil
 }
 
-func validateMessageRuleNode(node *MessageRuleDependencyGraphNode) error {
-	if err := validateMessageRuleNodeCyclicDependency(node, node.Children); err != nil {
+func validateMessageRuleNode(node *MessageRuleDependencyGraphNode) *LocationError {
+	if err := validateMessageRuleNodeCyclicDependency(node, make(map[*MessageRuleDependencyGraphNode]struct{}), []*MessageRuleDependencyGraphNode{}); err != nil {
 		return err
 	}
 	return nil
 }
 
-func validateMessageRuleNodeCyclicDependency(target *MessageRuleDependencyGraphNode, children []*MessageRuleDependencyGraphNode) error {
-	for _, child := range children {
-		if target == child {
-			return fmt.Errorf(`found cyclic dependency in "%s.%s" message`, target.Message.PackageName(), target.Message.Name)
+func validateMessageRuleNodeCyclicDependency(target *MessageRuleDependencyGraphNode, visited map[*MessageRuleDependencyGraphNode]struct{}, path []*MessageRuleDependencyGraphNode) *LocationError {
+	path = append(path, target)
+	if _, exists := visited[target]; exists {
+		var messages []string
+		for _, node := range path {
+			messages = append(messages, node.Message.Name)
 		}
-		if err := validateMessageRuleNodeCyclicDependency(target, child.Children); err != nil {
-			return err
+		dependencyPath := strings.Join(messages, " => ")
+
+		msg := target.BaseMessage
+		for idx, dep := range msg.Rule.MessageDependencies {
+			if dep.Message == target.Message {
+				return ErrWithLocation(
+					fmt.Sprintf(
+						`found cyclic dependency for "%s.%s" message in "%s.%s. dependency path: %s"`,
+						target.Message.PackageName(), target.Message.Name,
+						msg.PackageName(), msg.Name, dependencyPath,
+					),
+					source.MessageDependencyMessageLocation(msg.File.Name, msg.Name, idx),
+				)
+			}
 		}
-		if err := validateMessageRuleNodeCyclicDependency(child, child.Children); err != nil {
+		return ErrWithLocation(
```

Comment:
I think this path isn't actually evaluated, but I would like to display the `depedencyPath` here as well .

Reply:
Ah, I was going to add it there as well but totally forgot it... 🤦‍♂️